src/com/hms_networks/americas/sc/taginfo/TagInfoManager.java

```diff
@@ -99,7 +99,9 @@ public static synchronized void refreshTagList() throws IOException {
 
     // Correct tag info array for gaps
     final int tagIdDiff = highestTagIdSeen - lowestTagIdSeen;
-    final int numTagIdGaps = tagIdDiff - IOManager.getNbTags();
+    // Indices start at 0 instead of 1. Account for the error when converting tag count to index.
```

Comment:
What indices are you referring to?

if:
highestTagIdSeen == 15
lowestTagIdSeen == 3
IOManager.getNbTags() == 4

then:
numTagIdGaps == (15-3) - 4  == 8

| 3 | 4 |  5 |  6 | 7 | 8 |  9 |10| 11|12 | 13 |
|---|---|---|---|---|---|---|---|---|---|---|
| x  | x | 1  | 2 |  3 | 4 | 5  | 6 | 7 | x | 8 | x |

x represents a existing tag

but for your proposed algorithm it would be 9?

Reply:
I made edits to this comment